libs/fluid-elements/provider/src/lib/provider.ts

```diff
@@ -0,0 +1,232 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  html,
+  property,
+  TemplateResult,
+  css,
+  customElement,
+  LitElement,
+  CSSResult,
+  unsafeCSS,
+} from 'lit-element';
+
+import * as designTokens from '@dynatrace/fluid-design-tokens';
+
+/** Defines the possible layout density options. */
+export type FluidLayoutDensity = 'default' | 'dense' | 'loose';
+
+/** Defines the available themes. */
+export type FluidTheme = 'abyss' | 'surface';
+
+/** Type of the design tokens module */
+export type FluidDesignTokens = typeof designTokens;
+
+const supportsAdoptedStylesheets =
+  'adoptedStyleSheets' in window.ShadowRoot.prototype;
+
+// The provider itself cannot extend FluidElement
+// since it would lead to a circular dependency.
+@customElement('fluid-provider')
```

Comment:
name is maybe a bit too generic - what does it provide? 
fluid-design-system-provider is maybe a better name - though a bit long